DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRule.cs

```diff
@@ -0,0 +1,55 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public abstract class AnonymizerRule
+    {
+        private readonly IAnonymizerProcessor _processor;
+
+        public AnonymizerRule(string method, string description, JObject ruleSetting = null, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+        {
+            EnsureArg.IsNotNull(method, nameof(method));
+            EnsureArg.IsNotNull(description, nameof(description));
+
+            Description = description;
+            processorFactory ??= new DicomProcessorFactory();
```

Comment:
>processorFactory [](http://example.com/codeflow?start=12&length=16)

this is same as settingFactory I mentioned in other comment. You can either create a singleton in its caller or you can create a new one and remove it from parameters.

Reply:
after revisit the DicomProcessorFactory code, you can simply change it to a static one and call it here. No need to pass it from the parameter....